ls1intum/Artemis
# Feature/quiz exercises/match percentage option for text fields

### Checklist

### Motivation and Context


https://github.com/ls1intum/Artemis/issues/2570

### Description

When creating a short answer quiz question, the user now has the option to select a match percentage between the sample answer and the answer provided (from 50% to 100%). If the provided answer is above that value, it is considered correct. The user can also select whether the letter case matters when comparing the answers.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercises and create a short answer quiz question.
4. Open a database GUI/MySQL CLI  and check if the "match letter case" and the "similarity value" get persisted correctly for different values in the "quiz_question" table.
5. Test if the quiz submissions for different combinations of "match_letter_case" and "similarity_value" work correctly.
For Example:
a) match_letter_case = true, similarity_value = 100, solution = "Example solution" shouldn't match "example solution"
b) match_letter_case = false, similarity_value = 100, solution = "Example solution" should match "example solution", but not "example sluton"
c) match_letter_case = false, similarity_value = 70, solution = "Example solution" should match "Exmaple solut", but not "Exmp sloti"
d) ...

### Test Coverage


### Screenshots


![image](https://user-images.githubusercontent.com/63168603/104818780-6ae60280-5832-11eb-8bce-509bcdb680b1.png)
![image](https://user-images.githubusercontent.com/63168603/104818799-7c2f0f00-5832-11eb-91bc-c9ff11efe81c.png)
![image](https://user-images.githubusercontent.com/63168603/104818839-a7196300-5832-11eb-9d64-8c2b8f713e19.png)
